### PR TITLE
RDISCROWD-6676: Record priv response under v2 staging buckets.

### PR DESCRIPTION
**Pull request by dchhabda**
* Record priv response under v2 staging buckets.
* Add botocore dependency for boto3

### CODE REVIEW

File: pybossa/api/task_run.py [hunk 1]
```diff
@@ -84,7 +85,7 @@ def _preprocess_post_data(self, data):
             data['info'] = {}
             pyb_answer_url = upload_json_data(json_data=info, upload_path=path,
                 file_name='pyb_answer.json', encryption=with_encryption,
-                conn_name='S3_TASKRUN', upload_root_dir=upload_root_dir
+                conn_name=conn_name, upload_root_dir=upload_root_dir
```
Comment (kbecker42) [inline pybossa/api/task_run.py:88]: Should a copy be made to both buckets for a period of time (6 months or a year) until final migration to v2?

File: pybossa/api/task_run.py [hunk 2]
```diff
@@ -68,6 +68,7 @@ class TaskRunAPI(APIBase):
     def _preprocess_post_data(self, data):
         with_encryption = app.config.get('ENABLE_ENCRYPTION')
         upload_root_dir = app.config.get('S3_UPLOAD_DIRECTORY')
+        conn_name = "S3_TASKRUN_V2" if app.config.get("S3_CONN_TYPE_V2") else "S3_TASKRUN"
```
Comment (kbecker42) [inline pybossa/api/task_run.py:71]: Consider defining the related variables in the class `TaskRunAPI` so all methods have access to it from a single location.

```python
class TaskRunAPI(APIBase):
   conn_name = "S3_TASKRUN_V2" if app.config.get("S3_CONN_TYPE_V2") else "S3_TASKRUN"
   bucket = app.config.get("S3_BUCKET_V2") if app.config.get("S3_CONN_TYPE_V2") else app.config.get("S3_BUCKET")
```

or

```python
class TaskRunAPI(APIBase):
   version_str = "_V2" if app.config.get("S3_CONN_TYPE_V2") else ""
   conn_name = "S3_TASKRUN" + version_str
   bucket_name = "S3_BUCKET" + version_str
   bucket = app.config.get(bucket_name)
```

**Review by kbecker42 — APPROVED** [review]
Thank you.

File: pybossa/api/task_run.py (hunk 1 above)
Comment (dchhabda) [reply pybossa/api/task_run.py:88]: As we're upgrading staging buckets to v2 and taskrun payloads contains url path referring to the staging bucket path, its not required to make a copy to v1 bucket.

File: pybossa/api/task_run.py (hunk 2 above)
Comment (dchhabda) [reply pybossa/api/task_run.py:71]: Make sense. I'll update the code. Thanks!

**Review by n00rsy — APPROVED** [review]
lgtm

**Pull request merged**